CONTRIBUTING.md

```diff
@@ -0,0 +1,30 @@
+# How to successfully make changes to Ninja
+
+We're very wary of changes that increase the complexity of Ninja (in particular,
+new build file syntax or command-line flags) or increase the maintenance burden
+of Ninja. Ninja is already successfully used by hundreds of developers for large
+projects and it already achieves (most of) the goals we set out for it to do.
+It's probably best to discuss new feature ideas on the
+[mailing list](https://groups.google.com/forum/#!forum/ninja-build) or in an
+issue before creating a PR.
+
+## Coding guidelines
+
+Generally it's the
+[Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html) with
+a few additions:
+
+* Ninja should also compile as C++03. You may use C++11 features in a test or an
+  unimportant tool if you guard your code with `#if __cplusplus > 201103L`.
+* We have used `using namespace std;` a lot. Please try to avoid it whenever
+  possible and use `std::`, but don't unnecessarely touch existing code.
+* All source files should have the Google Inc. license header.
```

Comment:
This isn't something new but taken from HACKING.md, so unrelated to this PR.

Reply:
Sorry, I disagree. You're adding it to a newly created file, which implies some legitimacy to the guideline.

If you don't want to change it in this PR, then I believe this PR should depend on the future PR that does change it, and not be merged until then.

Reply:
I can rename HACKING.md with Git so that this line isn't touched. Would that be better?

I also don't want to change it btw.

Reply:
Yes, renaming the file would be better.

Do you mean you don't want to be the one to change it?

Or do you mean that you want the instruction to inject Google's copyright, illegitimately, into every file contributed by non-google employees, to stay in the guidelines?

In either situation, is there a way I can convince you?

Reply:
> Do you mean you don't want to be the one to change it?

I mean that I don't want to change it in general.

> In either situation, is there a way I can convince you?

No, I don't think so. It would be quite a political discussion to be held, which I'm not good at in English and also not interested in.

Reply:
Fine.

Who do I need to talk to then?

It's incredibly inappropriate for the Ninja project to have this policy. The most polite that I can be about it is that the Ninja project is asking people to lie about who owns the code. Lying is wrong.

Reply:
I resolved the conversation after my first response, sorry that it came off wrong.

Reply:
Who do I need to talk to?

Reply:
No idea.

Reply:
@nico @evmar

Reply:
Nico and Evmar have no opinion on the subject apparently.

I'm telling you that requiring this specific copyright line in every file is legally problematic, and ethically unsound.

But it's clear that nothing is going to change, so I'm not going to expend further effort to address the problem.